src/se/terhol/test34/HouseImpl.java

```diff
@@ -0,0 +1,92 @@
+package se.terhol.test34;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.Set;
+
+/**
+ * @author Tereza Holm
+ */
+public class HouseImpl implements House {
+    Map<Room, Set<Room>> myHouse = new HashMap<>();
+    Map<Person, Room> currentHouse = new HashMap<>();
+
+    public HouseImpl(String input) throws IOException {
+        InputStream is = Files.newInputStream(Paths.get(input));
+        this.loadPlan(is);
+    }
+
+    @Override
+    public void placePerson(Person person, Room room) {
+        if (person == null) {
+            throw new IllegalArgumentException();
+        }
+        currentHouse.put(person, room);
+    }
+
+    @Override
+    public Room whereIs(Person person) {
+        return currentHouse.get(person);
+    }
+
+    @Override
+    public void enterNextDoor(Person person, Room room) {
+        if (!currentHouse.containsKey(person) || !this.isNext(currentHouse.get(person), room)) {
+            throw new IllegalArgumentException();
+        }
+        this.placePerson(person, room);
+    }
+
+    @Override
+    public void savePlan(OutputStream os) {
+        try (BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(os))) {
+            for (Room room : myHouse.keySet()) {
+                for (Room roomNext : myHouse.get(room)) {
+                    writer.write(room.toString() + ":" + roomNext.toString());
```

Comment:
It is much better to use `String.format()` to output the text instead of String concatenation. 

Material to study: https://dzone.com/articles/java-string-format-examples